go/proof.go

```diff
@@ -163,6 +168,95 @@ func (p *ExistenceProof) calculate(spec *ProofSpec) (CommitmentRoot, error) {
 	return res, nil
 }
 
```

Comment:
I think that once we merge in your documentation in IBC, and even potentially add the mermaid diagram in [1], we should link to it here as a reference

[1] https://github.com/pokt-network/pocket/pull/845#discussion_r1245881510